Makefile

```diff
@@ -515,10 +515,14 @@ endif
 
 # Make tarball with only Julia code
 light-source-dist: light-source-dist.tmp
-	# Prefix everything with the current directory name (usually "julia"), then create tarball
-	DIRNAME=$$(basename $$(pwd)); \
+	# Prefix everything with "julia-$(commit-sha)/" or "julia-$(version)/" and then create tarball
+	# To achieve prefixing, we temporarily create a symlink in the source directory that points back
+	# to the source directory.
+	DIRNAME=julia-$(JULIA_COMMIT); \
 	sed -e "s_.*_$$DIRNAME/&_" light-source-dist.tmp > light-source-dist.tmp1; \
-	cd ../ && tar -cz --no-recursion -T $$DIRNAME/light-source-dist.tmp1 -f $$DIRNAME/julia-$(JULIA_VERSION)_$(JULIA_COMMIT).tar.gz
+	ln -s . $$DIRNAME || exit 1; \
```

Comment:
Why `|| exit 1`?   I'm not sure I see why that would change any outcome?

Reply:
I always assumed that just `exit` would just return `0`, but it turns out that is not the case -- it returns `$?`. Removed the `1`s.

Reply:
`exit 1` explicitly causes `bash` to exit with return code `1`, so I think what you want is `exit 0`.

Reply:
Actually, I'm not sure that `exit 0` is what you want.  What are you trying to do?  Do you want to ignore `ln -s` errors?  Or do you want `ln -s` errors to prevent running the `tar` command?  Or do you want `ln -s` errors to run the `tar` command but prevent the `make` build from finishing?

Reply:
I think the best thing to do then is to not have an `exit` at all, and replace `ln -s . || exit 1; \` with `ln -s . && \`

Reply:
I also need to connect `tar` and `rm` with `&&` then. And could connect `sed` too -- unlikely to fail, but might as well check.

Reply:
It’s a Makefile—it’s already checking the exit code of each bash process (unless prefixed with -). No need to duplicate that.

Reply:
Also, `|| exit` is the same as `|| true`, since the default exit code is success